crates/solvers/src/api/routes/metrics.rs

```diff
@@ -0,0 +1,9 @@
+use prometheus::Encoder;
+
+pub async fn metrics() -> String {
+    let registry = observe::metrics::get_registry();
+    let encoder = prometheus::TextEncoder::new();
+    let mut buffer = Vec::new();
+    encoder.encode(&registry.gather(), &mut buffer).unwrap();
+    String::from_utf8(buffer).unwrap()
```

Comment:
Looks like this code is the only reason we have to depend on `prometheus`. Maybe it makes sense to shared this implementation in the `observe` crate which already exports some functions related to metrics.